internal/pkg/manifest/env.go

```diff
@@ -494,63 +501,40 @@ func (cfg *EnvironmentConfig) ELBAccessLogs() (*ELBAccessLogsArgs, bool) {
 	return &accessLogs.AdvancedConfig, true
 }
 
-// ALBSecurityGroupsConfig represents security group configuration settings for an ALB.
-type ALBSecurityGroupsConfig struct {
-	Ingress Ingress `yaml:"ingress"`
-}
-
-func (cfg ALBSecurityGroupsConfig) IsEmpty() bool {
-	return cfg.Ingress.IsEmpty()
-}
-
-// Ingress represents allowed ingress traffic from specified fields.
-type Ingress struct {
-	RestrictiveIngress RestrictiveIngress `yaml:"restrict_to"`
-	VPCIngress         *bool              `yaml:"from_vpc"`
-}
-
-// ALBIngressRestrictedToCDN returns true when the environment is configured
-// to only allow ALB ingress from the CDN.
-func (cfg *EnvironmentConfig) ALBIngressRestrictedToCDN() bool {
-	return aws.BoolValue(cfg.HTTPConfig.Public.SecurityGroupConfig.Ingress.RestrictiveIngress.CDNIngress)
-}
-
 // RestrictiveIngress represents ingress fields which restrict
 // default behavior of allowing all public ingress.
 type RestrictiveIngress struct {
 	CDNIngress *bool `yaml:"cdn"`
 }
 
+// RelaxedIngress contains ingress configuration to add to a security group.
+type RelaxedIngress struct {
+	VPCIngress *bool `yaml:"from_vpc"`
+}
+
+// IsEmpty returns true if there are no specified fields for relaxed ingress.
+func (i RelaxedIngress) IsEmpty() bool {
+	return i.VPCIngress == nil
+}
+
 // IsEmpty returns true if there are no specified fields for restrictive ingress.
 func (i RestrictiveIngress) IsEmpty() bool {
 	return i.CDNIngress == nil
 }
 
-// IsEmpty returns true if there are no specified fields for ingress.
-func (i Ingress) IsEmpty() bool {
-	return i.VPCIngress == nil && i.RestrictiveIngress.IsEmpty()
-}
-
 // IsEmpty returns true if there is no customization to the public ALB.
 func (cfg PublicHTTPConfig) IsEmpty() bool {
-	return len(cfg.Certificates) == 0 && cfg.SecurityGroupConfig.IsEmpty() && cfg.ELBAccessLogs.isEmpty()
+	return len(cfg.Certificates) == 0 && cfg.DeprecatedSG.IsEmpty() && cfg.ELBAccessLogs.isEmpty()
 }
 
 type privateHTTPConfig struct {
-	InternalALBSubnets   []string             `yaml:"subnets,omitempty"`
-	Certificates         []string             `yaml:"certificates,omitempty"`
-	SecurityGroupsConfig securityGroupsConfig `yaml:"security_groups,omitempty"`
+	InternalALBSubnets   []string                          `yaml:"subnets,omitempty"`
+	Certificates         []string                          `yaml:"certificates,omitempty"`
+	SecurityGroupsConfig DeprecatedALBSecurityGroupsConfig `yaml:"security_groups,omitempty"` // Deprecated. This field is now available in Ingress.
```

Comment:
nit: for consistency with public http config:
```suggestion
	DeprecatedSG DeprecatedALBSecurityGroupsConfig `yaml:"security_groups,omitempty"` // Deprecated. This field is now available in Ingress.
```

Reply:
I just figured that out and was sending out the commit before you see it 😁